src/management/projects.ts

```diff
@@ -90,6 +95,78 @@ export async function createFirebaseProjectAndLog(
   }
 }
 
+/**
+ * Get the user's desired project, prompting if necessary.
+ */
+export async function getOrPromptProject(options: any): Promise<FirebaseProjectMetadata> {
+  if (options.project) {
+    return await getFirebaseProject(options.project);
+  }
+  return selectProjectInteractively();
+}
+
+async function selectProjectInteractively(
+  pageSize: number = MAXIMUM_PROMPT_LIST
+): Promise<FirebaseProjectMetadata> {
+  const { projects, nextPageToken } = await getProjectPage(pageSize);
+  if (projects.length === 0) {
+    throw new FirebaseError("There is no Firebase project associated with this account.");
+  }
+  if (nextPageToken) {
+    // Prompt user for project ID if we can't list all projects in 1 page
+    return selectProjectByPrompting();
+  }
+  return selectProjectFromList(projects);
+}
+
+async function selectProjectByPrompting(): Promise<FirebaseProjectMetadata> {
+  const projectId = await promptOnce({
+    type: "input",
+    message: "Please input the project ID you would like to use:",
+  });
+
+  return await getFirebaseProject(projectId);
+}
+
+/**
+ * Presents user with list of projects to choose from and gets project information for chosen project.
+ */
+async function selectProjectFromList(
+  projects: FirebaseProjectMetadata[] = []
+): Promise<FirebaseProjectMetadata> {
+  let choices = projects.filter((p: FirebaseProjectMetadata) => !!p).map((p) => {
+    return {
+      name: p.projectId + (p.displayName ? ` (${p.displayName})` : ""),
+      value: p.projectId,
+    };
+  });
+  choices = _.orderBy(choices, ["name"], ["asc"]);
+
+  if (choices.length >= 25) {
+    utils.logBullet(
+      `Don't want to scroll through all your projects? If you know your project ID, ` +
+        `you can initialize it directly using ${clc.bold(
+          "firebase init --project <project_id>"
+        )}.\n`
+    );
+  }
+  const projectId: string = await promptOnce({
+    type: "list",
+    name: "id",
+    message: "Select a default Firebase project for this directory:",
+    choices,
+  });
+
+  let project: FirebaseProjectMetadata | undefined;
```

Comment:
I believe this typing is unnecessary.

`Array.prototype.find` returns an object in the array, or undefined, by definition. Since the `projects` array is typed, doing `const project = projects.find(...)` should give you the correct type (then check for it not existing and return project as is already done - this single line is redundant)

Reply:
Done